src/reporters/composite_reporter.js

```diff
@@ -33,24 +33,50 @@ export default class CompositeReporter {
     }
 
     report(span: Span): void {
-        for (let i = 0; i < this._reporters.length; i++) {
-            let reporter = this._reporters[i];
-            reporter.report(span);
-        }
+        this._reporters.map((r) => {
+            r.report(span);
+        });
```

Comment:
Popular opinion just elected Trump. I only trust benchmarks.

If functional style is faster that for-loop, then at least this should be `forEach`, not `map`, because map is used to transform collection into another list, but here we're not interested in any return values, so `forEach` (which returns void) is the correct monad to use.